pkg/argocd/update_test.go

```diff
@@ -1538,20 +1536,6 @@ replicas: 1
 		require.NoError(t, err)
 		assert.NotEmpty(t, yaml)
 		assert.Equal(t, strings.TrimSpace(strings.ReplaceAll(expected, "\t", "  ")), strings.TrimSpace(string(yaml)))
-
-		// when image.spec.foo fields are missing in the target helm value file,
-		// they should be auto created without corrupting any other pre-existing elements.
-		originalData = []byte("test-value1: one")
-		expected = `
-test-value1: one
-image:
-  spec:
-    foo: nginx:v1.0.0
-`
-		yaml, err = marshalParamsOverride(&app, originalData)
-		require.NoError(t, err)
-		assert.NotEmpty(t, yaml)
-		assert.Equal(t, strings.TrimSpace(strings.ReplaceAll(expected, "\t", "  ")), strings.TrimSpace(string(yaml)))
```

Comment:
Any reason this part of the test is removed? I think we should keep any existing test as much as we can, and add new tests for introduced new features. 

The above test verifies that 
* the updater can infer the correct param name (whether to interpret dotted names as nested or flat ones) based on the existing format in use. 
* any added new parameters will not corrupt or interfere with any existing parameters.